examples/cart/index.js

```diff
@@ -114,44 +157,80 @@ $(function() {
 
   /* Attach and control listeners onto buy button
   ============================================================ */
-  function attachBuyButtonListeners(product) {
-    $('.buy-button').on('click', function (event) {
-      event.preventDefault();
-      var id = product.selectedVariant.id;
-      addVariantToCart(product.selectedVariant, 1);
-    });
+  function buyButtonClickHandler(evt) {
+    evt.preventDefault();
+    var id = product.selectedVariant.id;
+    var quantity;
+    var cartLineItem = findCartItemByVariantId(id);
+    if (cartLineItem) {
+      quantity = cartLineItem.quantity + 1;
+    } else {
+      quantity = 1;
+    }
+    addOrUpdateVariant(product.selectedVariant, quantity);
+    setPreviousFocusItem(evt.target);
+    $('#checkout').focus();
   }
 
-  /* Increase product variant quantity in cart
+  /* Decrease product variant quantity in cart
   ============================================================ */
-  function attachQuantityIncrementListeners(product) {
-    $('.cart').on('click', '.quantity-increment', function() {
-      var variantId = parseInt($(this).attr('data-variant-id'), 10);
-      var variant = product.variants.filter(function (variant) {
-        return (variant.id === variantId);
-      })[0];
-
-      $(this).closest('.cart-item').addClass('js-working');
-      $(this).attr('disabled', 'disabled');
-
-      addVariantToCart(variant, 1);
-    });
+  function decrementQuantity(evt) {
+    var variantId = parseInt($(this).attr('data-variant-id'), 10);
+    var variant = product.variants.filter(function (variant) {
+      return (variant.id === variantId);
+    })[0];
+    var quantity;
+    var cartLineItem = findCartItemByVariantId(variant.id);
+    if (cartLineItem) {
+      quantity = cartLineItem.quantity - 1;
+      updateVariantInCart(cartLineItem, quantity);
+    }
   }
 
-  /* Decrease product variant quantity in cart
+  /* Increase product variant quantity in cart
   ============================================================ */
-  function attachQuantityDecrementListeners(product) {
-    $('.cart').on('click', '.quantity-decrement', function() {
-      var variantId = parseInt($(this).attr('data-variant-id'), 10);
-      var variant = product.variants.filter(function (variant) {
-        return (variant.id === variantId);
-      })[0];
+  function incrementQuantity(evt) {
+    var variantId = parseInt($(this).attr('data-variant-id'), 10);
+    var variant = product.variants.filter(function (variant) {
+      return (variant.id === variantId);
+    })[0];
+    var quantity;
+    var cartLineItem = findCartItemByVariantId(variant.id);
+    if (cartLineItem) {
+      quantity = cartLineItem.quantity + 1;
+      updateVariantInCart(cartLineItem, quantity);
+    }
+  }
```

Comment:
I can think of a couple ways we could combine the increment/decrement/input quantity methods. One way would be to have an `updateQuantity` function which accepts a function as a parameter. That function would accept the current value and return the new value. Then in your `updateQuantity` function you could say 
```
var quantity = fn(currentQuantity)
```

and then call it with like
```
updateQuantity(function (current) {
   return current + 1;
});
```

Or: 

```
var addOne = function (current) {return current + 1}
var incrementQuantity = updateQuantity.bind(addOne);
```